### PR TITLE
[RDY] pythonPackages.cmd2: 0.7.7 -> 0.8.0

### PR DESCRIPTION
**Pull request by teto**
Some cool stuff like history saving to transcripts, support for argparse
based parsers etc:
https://github.com/python-cmd2/cmd2/blob/master/CHANGELOG.md

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/cmd2/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{stdenv, fetchurl, buildPythonPackage, pyperclip, six, pyparsing}:
+buildPythonPackage rec {
+  pname = "cmd2";
+  version = "0.8.0";
+
+  src = fetchurl {
```
Comment (dotlambda) [inline pkgs/development/python-modules/cmd2/default.nix]: Please use `fetchPypi` instead.

File: pkgs/development/python-modules/cmd2/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  propagatedBuildInputs = [
+    pyperclip
+    six
+    pyparsing
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Enhancements for standard library's cmd module";
+    homepage = http://packages.python.org/cmd2/;
```
Comment (dotlambda) [inline pkgs/development/python-modules/cmd2/default.nix]: 404

File: pkgs/development/python-modules/cmd2/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+buildPythonPackage rec {
+  pname = "cmd2";
+  version = "0.8.0";
+
+  src = fetchurl {
+    url = "mirror://pypi/c/cmd2/${pname}-${version}.tar.gz";
+    sha256 = "1fmpn3dwc0pyhdi6q877mb8i4vw9p9a0lgmrvsaclfah62grsfjb";
+  };
+
+  # No tests included
```
Comment (dotlambda) [inline pkgs/development/python-modules/cmd2/default.nix]: On GitHub, there is a tests directory.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by teto** [comment]
I had copy/pasted from the old derivation. I updated it but still don't run the tests. They are tox based which IIRC won't work on nixos.

**Comment by dotlambda** [comment]
Maybe, you have luck running the tests with
```
checkPhase = ''
  py.test
'';
```

**Comment by teto** [comment]
wow, I 've seen some people asking on IRC and myself trying to pull tox for it to fail. This was able to run the tests but 25 out 191 failed so I still disabled them. I might try to fix some along the way as I develop my app against cmd2 and so far it works great with 0.8 (and I use quite a few features).

**Comment by dotlambda** [comment]
Tox just runs some command, usually `py.test`, `python -m unittest` or something like that. You can find out which one by looking at the `tox.ini` file.

**Comment by dotlambda** [comment]
@teto Maybe you could open an upstream issue about the failing tests.

**Review by dotlambda — APPROVED** [review]

**Comment by Mic92** [comment]
@GrahamcOfBorg  pythonPackages.cmd2

**Comment by Mic92** [comment]
@GrahamcOfBorg build pythonPackages.cmd2

**Comment by Mic92** [comment]
@GrahamcOfBorg build python2.pkgs.cmd2 python3.pkgs.cmd2

**Comment by Mic92** [comment]
@FRidh is a test that takes over an hour on aarch64 reasonable for this kind of package?

**Comment by teto** [comment]
@Mic92 should we conditionally add pbpaste on darwin and rerun the tests ? I don't have a mac so I would rather let someone do it.

**Comment by Mic92** [comment]
@GrahamcOfBorg build python2.pkgs.cmd2 python3.pkgs.cmd2

**Comment by Mic92** [comment]
mac os sandbox was disabled. Maybe it finds pbpaste now.

**Comment by Mic92** [comment]
Pandas fakes those on darwin: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/pandas/default.nix#L75

**Comment by Mic92** [comment]
@GrahamcOfBorg build python2.pkgs.cmd2

**Comment by Mic92** [comment]
@GrahamcOfBorg build python2.pkgs.cmd2

**Comment by Mic92** [comment]
@GrahamcOfBorg build python2.pkgs.cmd2

**Comment by Mic92** [comment]
finally

**Comment by teto** [comment]
thanks for the patches !

**Pull request merged**